kmj - 사다리 게임

jyami-kim opened the pull request.

jyami-kim committed:
init

jyami-kim committed:
setting initial design

jyami-kim committed:
ladder game first

jyami-kim committed:
get pillar's bridge location list

jyami-kim committed:
create all bridges of pillars

jyami-kim committed:
create ladder

jyami-kim committed:
ladder Game

jyami-kim committed:
ladder game complete

pci2676 commented on src/main/java/LadderGame.java:
```diff
@@ -1,4 +1,18 @@
 
+    public static void main(String args[]) {
+        InputView inputView = new InputView(System.in);
+        InputData inputData = new InputData(inputView.participants(), inputView.ladderHeight());
```
InputData inputData = inputView.getInputData();
같은 형식이 나는 더 읽기 좋은것 같아

pci2676 commented on src/main/java/data/InputData.java:
```diff
@@ -0,0 +1,33 @@
+    private String participants;
+
+    public InputData(String participants, int ladderHeight) {
+        this.ladderWidth = ladderWidth(participants);
```
굳이 ladderWidth값을 들고있지 않고
participants 에서 split한거 length 반환하는 거 로 width 반환해도 되지 않을까 싶어
그러면 멤버변수 수도 줄일수 있으니까

pci2676 commented on src/main/java/data/InputData.java:
```diff
@@ -0,0 +1,33 @@
+
+    public InputData(String participants, int ladderHeight) {
+        this.ladderWidth = ladderWidth(participants);
+        this.ladderHeight = ladderHeight;
```
입력받은 Height값이 유효한지 체크하는 함수 있으면 좋을것 같아

pci2676 commented on src/main/java/domain/ladder/Pillar.java:
```diff
@@ -0,0 +1,98 @@
+import static util.Util.createRandomIntegers;
+import static util.Util.createRandomIntegersWithRestriction;
+
+public class Pillar {
```
Pillar 클래스에서 하는일이 너무 많은거같아
아래 Bridge만드는 로직을 다른곳으로 빼는게 난 좋을거같아

jyami-kim committed:
after advise

chldbtjd2272 commented on src/main/java/domain/ladder/Pillar.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+
+    private boolean isLastPillar(GameStartOption gameStartOption, Pillar previousPillar) {
```
return gameStartOption.getLadderWidth() - previousPillar.getPillarNum() == 2

chldbtjd2272 commented on src/main/java/domain/ladder/Pillar.java:
```diff
@@ -0,0 +1,92 @@
+        return false;
+    }
+
+    private boolean isFirstPillar(Pillar previousPillar) {
```
return Objects.isNull(previousPillar)

chldbtjd2272 commented on src/main/java/domain/ladder/Pillar.java:
```diff
@@ -0,0 +1,92 @@
+    }
+
+    private List<Bridge> createLeftRightBridges(Pillar previous, Integer height) {
+        List<Integer> previousLocations = previous.getBridgesDirectionLocation(LinkedType.RIGHT);
```
오른쪽 위치 찾고 밑에서 createLeft할때 왜 또 다시찾아?

chldbtjd2272 commented on src/main/java/domain/ladder/Pillar.java:
```diff
@@ -0,0 +1,92 @@
+        return false;
+    }
+
+    private List<Bridge> createLeftRightBridges(Pillar previous, Integer height) {
```
이부분 너무 복잡해서 다시 정리해 보는게 좋을거같아

jyami-kim committed:
constructor edit

jyami-kim committed:
constructor refactoring

jyami-kim committed:
ladder output domain

jyami-kim committed:
ladder game complete

jyami-kim committed:
Merge branch 'kmj' into master

jyami-kim committed:
Merge pull request #8 from mjung1798/master

kmj - 사다리게임

jyami-kim committed:
merging

jyami-kim committed:
[ 수정완료 ]
1. package 설계 다시하기

각 List, List 생성부분 maker라는 class로 분리하기
throw error 처리
factory method 패턴 : PillarFactory
test code 수정
2. output view 수정

bridge와 높이 들고있는 dto로 만들기 (가로로 출력할 수 있는 dto로 만들기)
converter class로 분리 : bridge 1개와 level 들고있는
찍기 편한 형태로 변환 (flatMap)
3. 마지막 요구사항 추가하기

실행결과
유저와 실행 결과 매칭해서 output 보여주기 - user class에서 calc 메소드 만들어서 구현
사다리 input output 결과 매칭하기

jyami-kim committed:
Merge branch 'master' of https://github.com/mjung1798/ladder-game

jyami-kim committed:
Merge branch 'master' into kmj
